src/Solnet.Rpc/Utilities/ShortVectorEncoding.cs

```diff
@@ -41,9 +41,8 @@ internal static byte[] EncodeLength(int len)
                 output[cursor] = (byte)elem;
                 cursor += 1;
             }
-
             byte[] bytes = new byte[cursor + 1];
-            Array.Copy(output, 0, bytes, 0, cursor + 1);
+            Array.Copy(output.ToArray(), 0, bytes, 0, cursor + 1);

```

Comment:
Given that ToArray creates a new copy in the heap, use `output.CopyTo(output)` to prevent this.